packages/backend-common/src/middleware/authenticationHandler.ts

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright 2020 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { NextFunction, Request, RequestHandler, Response } from 'express';
+
+export interface Authenticator {
+  authenticateRequest: (request: Request) => boolean;
+
+  // log the failed request if needed
+  logRequest?: (request: Request) => void;
+}
+
+/**
+ * Express middleware to authenticate requests
+ *
+ * @returns An Express request handler
+ */
+export function authenticationHandler(
+  authenticator: Authenticator,
+): RequestHandler {
```

Comment:
I agree that the specific ones are the most beneficial additions. 

What I wanted to was establish a simple and generic interface that can be used to build specific middleware.

Currently at Zalando for example, we use an internal endpoint to which a token must be sent , to verify it, rather than having access to a list of published JWKS. 

Similarly, on the Frontend, I believe that applications are free to send Tokens to the backends as they see fit using the `oauth2ApiRef` for example. 

The interface I created can be implemented to support all such schemes. Perhaps to add more value, I can maybe add an implementation that uses the keys available in the `TokenIssuer` to verify the tokens generated by the `auth-backend`.